src/solver/prelu/backward_prelu_multi_weights.cpp

```diff
@@ -0,0 +1,180 @@
+                            "MIOpenReduceSum.cpp",
+                            "Reduce1dSum",
+                            build_params));
+    }
+
+    result.invoker_factory = [](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle_, const AnyInvokeParams& raw_params) {
+            decltype(auto) params = raw_params.CastTo<miopen::prelu::InvokeParams>();
+
+            auto elapsed = 0.f;
+            HipEventPtr start;
+            HipEventPtr stop;
+
+            bool reset_profiling_state = false;
+            if(handle_.IsProfilingEnabled())
+            {
+                reset_profiling_state = true;
+                handle_.EnableProfiling(false);
+                hipStreamSynchronize(handle_.GetStream());
+                start = miopen::make_hip_event();
+                stop  = miopen::make_hip_event();
+                hipEventRecord(start.get(), handle_.GetStream());
+            }
+
+            int kernelCnt = 0;
+
+            /* Phase 1: Calc gradient for each elements. */
+            {
+                auto input_tv         = get_inner_expanded_tv<VIEW_DIMS>(deref(params.inputDesc));
+                auto output_grad_tv   = get_inner_expanded_tv<VIEW_DIMS>(deref(params.doutputDesc));
+                auto input_grad_tv    = get_inner_expanded_tv<VIEW_DIMS>(deref(params.dinputDesc));
+                decltype(auto) kernel = handle_.Run(kernels[kernelCnt++]);
+                kernel(params.input,
+                       params.weight,
+                       params.doutput,
+                       params.dinput,
+                       params.workspace,
+                       static_cast<uint64_t>(deref(params.inputDesc).GetElementSize()),
+                       input_tv,
+                       output_grad_tv,
+                       input_grad_tv);
+            }
+
+            /* Phase 2: Reduce gradient. */
+            {
+                size_t output_numel = deref(params.weightDesc).GetElementSize();
+                size_t outer_size   = deref(params.inputDesc).GetLengths()[0];
+                size_t inner_size =
+                    deref(params.inputDesc).GetElementSize() / outer_size / output_numel;
+                handle_.Run(kernels[kernelCnt++])(
+                    params.workspace, params.dweight, output_numel, inner_size, outer_size);
+            }
+
+            if(reset_profiling_state)
+                handle_.EnableProfiling(true);
+            if(handle_.IsProfilingEnabled())
+            {
+                hipEventRecord(stop.get(), handle_.GetStream());
+                hipEventSynchronize(stop.get());
+                hipEventElapsedTime(&elapsed, start.get(), stop.get());
+
+                // Clean up
+                hipEventDestroy(start.get());
+                hipEventDestroy(stop.get());
+                handle_.ResetKernelTime();
+                handle_.AccumKernelTime(elapsed);
+            };
```

Comment:
The second part of profiling simplification
```suggestion
            if(profiling)
            {
                hipEventRecord(stop.get(), handle_.GetStream());
                handle_.EnableProfiling(true);
                hipEventSynchronize(stop.get());
                hipEventElapsedTime(&elapsed, start.get(), stop.get());

                // Clean up
                hipEventDestroy(start.get());
                hipEventDestroy(stop.get());
                handle_.ResetKernelTime();
                handle_.AccumKernelTime(elapsed);
            };
```

Reply:
Fixed it.